### PR TITLE
Compliance test unset interface state

### PR DESCRIPTION
**Pull request by emmurd**

### CODE REVIEW

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from hamcrest import assert_that, is_
+from netman.core.objects.exceptions import UnknownInterface
+
+from tests.adapters.configured_test_case import ConfiguredTestCase
+
+
+class UnsetInterfaceStateTest(ConfiguredTestCase):
+    _dev_sample = "juniper"
+
+    def test_unsetting_an_interface(self):
```
Comment (lindycoder) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: This is not readable : 

UnsetInterfaceState unsetting an interface

test name should be readable when appended after the method name, according to the readme

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        self.try_to.unset_interface_state(self.test_ports[0].name)
```
Comment (lindycoder) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: you're trying to unset and interface state after not setting it?

i also don't really see why you need 2 interfaces.

Wouldn't this test work out:

original_state = get state
set state ON
unset
assert new state == original
set state OFF
unset
assertt new state == original

**Comment by emmurd** [comment]
Comment addressed. Please review :)

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+from netman.core.objects.exceptions import UnknownInterface
+from netman.core.objects.interface_states import ON, OFF
+
+from tests.adapters.configured_test_case import ConfiguredTestCase
+
+
+class UnsetInterfaceStateTest(ConfiguredTestCase):
+    _dev_sample = "juniper"
+
+    def test_unset_an_interface_to_default_state(self):
```
Comment (lindycoder) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: test_returns_an_interface_to_default_state

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+        default_state = self.client.get_interface(self.test_port).shutdown
+
+        self.try_to.set_interface_state(self.test_port, ON)
+        self.try_to.unset_interface_state(self.test_port)
```
Comment (lindycoder) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: use client not try_to

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2015 Internap.
```
Comment (fbouliane) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: 2016 :D

**Comment by fbouliane** [comment]
:+1: for me other than the copyright.

File: tests/adapters/compliance_tests/unset_interface_state_test.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+from tests.adapters.configured_test_case import ConfiguredTestCase
+
+
+class UnsetInterfaceStateTest(ConfiguredTestCase):
+    _dev_sample = "juniper"
+
+    def test_returns_an_interface_to_default_state(self):
+
+        default_state = self.client.get_interface(self.test_port).shutdown
```
Comment (mat128) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: Shouldn't that member be called "state" if the action to change it is "{un,}set_interface_state"?

Not blocking the PR for this.

Comment (lindycoder) [inline tests/adapters/compliance_tests/unset_interface_state_test.py]: That would be part of another refactor, but i agree, issue?

**Pull request merged**